frontend/src/pages/Projects/Project/AgreementSection/handleOnSubmit.ts

```diff
@@ -0,0 +1,31 @@
+import { apiAxios } from "../../../../utils";
+
+export const handleOnSubmit = async (
+  query: { data: { data: { [x: string]: { value: string } | number } } },
+  values: { [x: string]: { value: string } }
+) => {
+  const changedValues = new Promise((resolve) => {
```

Comment:
just wondering if this can be created in a more modular way, seems like the same code being repeated.